### PR TITLE
ECC

### PR DESCRIPTION
**Pull request by OmegaPointZero**
## Description of changes in release / Impact of release:
(insert text here)

## Documentation
(insert text here)

### Is this a breaking change?
- [ ] Yes
- [ ] No

### If you answered Yes then describe why is it so
(insert text here if applicable)

### Is there a way to disable the change?
- [ ] Use previous release
- [ ] Use a feature flag
- [ ] No

#### Additional details go here
(insert text here if applicable)

### CODE REVIEW

**Comment by mahmoudimus** [comment]
@OmegaPointZero can you tell me a bit more about this PR?

**Comment by OmegaPointZero** [comment]
@mahmoudimus this was regarding ES256 with Jose, Drew asked me for a PR and test to see what I'd done since he was working on something with ECC as it was.

File: larky/src/main/resources/vendor/Crypto/Signature/DSS.star [hunk 1]
```diff
@@ -268,41 +268,153 @@ def DeterministicDsaSigScheme(key, encoding, order, private_key):
...
+
+        if self._encoding == 'binary':
+            if len(signature) != (2 * self._order_bytes):
+                fail("ValueError: The signature is not authentic (length)")
+            r_prime, s_prime = [Integer.from_bytes(x)
+                                for x in (signature[:self._order_bytes],
+                                          signature[self._order_bytes:])]
+        else:
+            # try:
+            der_seq = DerSequence().decode(signature, strict=True)
```
Comment (mahmoudimus) [inline larky/src/main/resources/vendor/Crypto/Signature/DSS.star:398]: to mimic tryp except:

```bzl
load("@vendor//option/result", Result=Result)

result = Result.Ok(DerSequence()).map(lambda x: x.decode(signature, strict=True))
der_seq = result.unwrap_or("ValueError: The signature is not authentic (DER content)")
# and continue down your merry business
```

**Review by aslepakurov — APPROVED** [review]

File: larky/src/main/resources/vendor/Crypto/Util/asn1.star [hunk 2]
```diff
@@ -867,7 +867,9 @@ def DerObjectId(value='', implicit=None, explicit=None):
         Raises:
             ValueError: in case of parsing errors.
         """
-
+        if type(der_encoded) == 'string':
```
Comment (mahmoudimus) [inline larky/src/main/resources/vendor/Crypto/Util/asn1.star:870]: @OmegaPointZero why was this put here? Was the DerObjectId not encoded property when it came in under decode()?

File: larky/src/main/resources/vendor/cryptography/hazmat/backends/pycryptodome.star [hunk 3]
```diff
@@ -722,6 +957,12 @@ def pycryptodome():
         return self._evp_pkey_to_private_key(evp_pkey)
 
     self.load_pem_private_key = load_pem_private_key
+
+    def load_pem_public_key(data):
+        if(re.search(r'-----BEGIN (EC|RSA) PRIVATE KEY-----', data)):
```
Comment (mahmoudimus) [inline larky/src/main/resources/vendor/cryptography/hazmat/backends/pycryptodome.star:962]: @OmegaPointZero can you help me understand why this is necessary? why wouldn't we call load_pem_private_key? Also, it looks like this is not what the cryptography library does... https://github.com/pyca/cryptography/blob/main/src/cryptography/hazmat/backends/openssl/backend.py#L920-L960

**Pull request merged**